Makefile

```diff
@@ -1,27 +1,48 @@
-PINT_VERSION='0.42.2'
+PINT_VERSION = 0.42.2
+PROMTOOL_VERSION = 2.42.0
+YQ_VERSION = 4.31.2
+EXTRACTED_RULE_FILE = test/promql/extracted-rules.yaml
+RULE_FILE = prometheus/base/prometheus.rules.yaml
 
 .PHONY: all
 all: prepare lint test_rules pint_lint
 
 .PHONY: prepare
-prepare: pint
-	./automation/promtool_tests.sh prepare
+prepare: pint promtool yq
+	echo "Extract Prometheus rules"
+	./yq ".spec" ${RULE_FILE} > ${EXTRACTED_RULE_FILE}
 
```

Comment:
Does this refactoring need any update in contribution guidelines ?

Reply:
You mean for how to run the tests locally? or you have something else in mind?

Reply:
Are you talking about this section?
```
### Prerequisites
* Linux machine with x86-64 architecture with bash installed.
* Basic tools: curl, tar, make.
```

Reply:
ok. In this case, no. There's no change. We've been using yq already. That section is just about the things that have to be pre-installed for `make all` to work. That haven't changed.

Reply:
Sorry for the confusion, https://github.com/redhat-appstudio/o11y/blob/main/CONTRIBUTING.md#automated-tests 
List of tools used for this project. Since we are installing yq within our make file. we can include it.

Reply:
I see. Those tools are ones that actually run tests for us. We're using yq for extracting the rules YAML rather than for actual testing.
Anyhow, this have not changed. Feel free to open a ticket for adding this if you think it should be added.